server/snapshot.go

```diff
@@ -219,11 +215,18 @@ func (c *channel) restoreFromSnapshot(snapshot io.ReadCloser) error {
 			fragmentPool.Put(fragment)
 		case spb.RaftSnapshotFragment_Subscription:
 			// Channel subscription.
+			acksPending := make(map[uint64]int64, len(fragment.Sub.AcksPending))
+			for _, seq := range fragment.Sub.AcksPending {
+				// Set 0 for expiration time. This will be computed
+				// when the follower becomes leader and attempts to
+				// redeliver messages.
+				acksPending[seq] = 0
+			}
 			sub := &subState{
 				SubState:    *fragment.Sub.State,
 				subject:     fragment.Sub.Subject,
 				ackWait:     computeAckWait(fragment.Sub.State.AckWaitInSecs),
-				acksPending: make(map[uint64]int64),
+				acksPending: acksPending,
```

Comment:
This is good for runtime, but you are not persisting those acks on store. After call to addSubscription, should you persist those acks or not?

Reply:
Yes, good catch. We need to persist them.